### ISSUE TITLE
Better delimiter highlighting

### ISSUE THREAD

**Issue by tinyCatzilla** [lervag/vimtex#2776]
As of right now, `texMathDelim` has some strange properties that I can't seem to wrangle.

- within any `texMathZone`, inequalities are recognized as delimiters and not math operators. 
- For commands that are seen "under-the-hood" without delimiters, the delimiters and arguments are not recognized. For example, using `\frac{1}{2}` instead of the shortcut `\frac12` will color the delimiters and numbers as part of the command. This sharply contrasts a similar fraction without a shortcut, like `\frac{12}{2}`.
- Paired delimiters are recognized, such as `\lvert` and `\rvert`. The unpaired delimiter '|' is also recognized. But the synonymous command for the unpaired delimiter \vert is only seen as a command. 

Below is a screenshot demonstrating these properties, where `texMathDelim` is pink, `texDelim` is dark blue, and `texCmd` and `texCmdEnv` are yellow.

![SCR-20230823-ngrr](https://github.com/lervag/vimtex/assets/60241608/d0b97c19-6105-499f-86ad-b598978c43d1)

In regards to inequalities being delimiters, this is something particularly strange to me. I haven't heard of <> being delimiters in LaTeX, but I could just be uninformed. Also, If any of these are fixable or problems on my end, I would appreciate support getting it working :).

Edit: fuck, sorry about closing and reopening. was trying to edit to show this other strange thing i realized with regards to math operators not being recognized when superscripted.
![SCR-20230823-nlqm](https://github.com/lervag/vimtex/assets/60241608/ece23208-5c22-409f-a259-521088f60af8)

**Comment by tinyCatzilla** [comment]
Oops, sorry

**Comment by lervag** [comment]
Are you sure you are not using Treesitter for highlighting? I believe things look different with VimTeX. See `:help vimtex-faq-treesitter`.

**Comment by tinyCatzilla** [comment]
I believe I disabled it with the following configuration. I also remember that I saw a VimTeX notification that treesitter was controlling highlighting before changing these options.

Is there a way to make sure that this is disabling treesitter as intended?

![SCR-20230824-pmmt](https://github.com/lervag/vimtex/assets/60241608/54d8d823-0ad7-4fbb-9691-a6ca8c06ed60)

**Comment by lervag** [comment]
Ok, sorry; I see that with default settings I can get the same result as you. So, there are several things going on here:

1. `\frac{1}{2}` is captured as a concealable symbol `½`. You can see it if you do `:set conceallevel=2`. You can disable conceals if you want with `let g:vimtex_syntax_conceal_disable = 1`.

2. `>` and `<` are both delimiters and operators. For no particular reason, I chose to highlight as delimiters. I guess it does make sense to highlight them as operators instead?

3. VimTeX does not support highlighting things like `\vert ... \vert` as delimiters, because they are not really that. If you use them as delimiters, you should always use `\lvert .. rvert`. Similar for things like `\bigl .. \bigr` and stuff. However, perhaps I'm being to strict here...

**Comment by Rmano** [comment]
My opinion:

1. I never use conceal, so I abstain here
3. I think `<` and `>` are more often used as operators than delimiters
4. I think you're right --- considering by default delimiter the ones with "direction"

**Comment by tinyCatzilla** [comment]
Thanks for the great support and confirmation that this is intended behavior. To respond with my opinion:

1. I also do not use conceal and thus abstain. However, I personally would find it nice if highlighting changed either by option or automatically to match the conceal option.
2. Yes, `<` and `>` not prioritized as operators is really my only major gripe. The rest is more-so to spark discussion.
3. I also think you're right; `\vert` should not be used as a delimiter (even if it is all-too-common). But then for consistency, neither should it's synonym`|`.

By the way, love the plugin -- keep up the great work.

**Comment by lervag** [comment]
Sorry for the long silence! I've had a lot of stuff going on, including silly stuff such as playing a computer game. It's a phase, it'll pass!

In any case:

1. `conceallevel` is not a global option, it is _windows local_. I believe using it to set the default value of `g:vimtex_syntax_conceal_disable` may yield an unexpected behaviour, and so I will keep these things as they are now. The conceal feature is the first section in `:help vimtex-syntax` and the current behaviour is explained at the top.

2. I agree and I've updated accordingly.

3. Good point about `|`. I've fixed that. I made some additional changes: `/` should also not be a delimiter, same for `.`.

This are minor things, but at the same time, I think they do make a noticable improvement. Thanks!

> By the way, love the plugin -- keep up the great work.

Thank you, I'm glad to hear it!
